### ISSUE TITLE
links object in POST response dirtying relationship

### ISSUE THREAD

**Issue by deepan83** [danielspaniel/ember-data-change-tracker#7]
Say I do a POST to create a photo with a photographer relationship,

```
POST /photos HTTP/1.1
Content-Type: application/vnd.api+json
Accept: application/vnd.api+json

{
  "data": {
    "type": "photos",
    "attributes": {
      "title": "Ember Hamster",
      "src": "http://example.com/images/productivity.png"
    },
    "relationships": {
      "photographer": {
        "data": { "type": "people", "id": "9" }
      }
    }
  }
}
```

The following JSON API compliant response to the POST makes the photographer relationship dirty (owing to the links object in the response),

```
{
  "data": {
    "id": 1,
    "type": "photos",
    "attributes": {
      "title": "Ember Hamster",
      "src": "http://example.com/images/productivity.png"
    },
    "relationships": {
      "photographer": {
        "links": {
           "self": "/api/photos/1/relationships/photographer",
           "related": ""/api/photos/1/photographer"" 
         },
        "data": { "type": "people", "id": "9" }
      }
    }
  }
} 
```

This means that a subsequent PATCH operation on this record (triggered by an update to title for example) will pass the photographer relationship in the request, despite the client not modifying it. Is there a way this can be avoided?

**Comment by danielspaniel** [comment]
What is the issue then .. are you using keepOnlyChanged mixin? and want to not resend this relationship?

**Comment by deepan83** [comment]
Yes, am using the mixin and prefer not to send this relationship

**Comment by danielspaniel** [comment]
ok .. you want to dig in there and find out why this is being flagged as changed?  or you want me to?

**Comment by deepan83** [comment]
Yeah am up for helping with this. Raised the issue to know if there was any existing workaround to this. Will do a PR once I get to the bottom of this.

**Comment by danielspaniel** [comment]
I have not seen this because I did not test with links payload .. so just needs a test or two to figure out what is going on. But I am not sure it is fixable though .. but if you dig in and see strange things, let me know. I will bring out the flashlight and goggles.

**Comment by ezekg** [comment]
@deepan83 Just for giggles, does the latest version fix this issue? It may have been related to #8.
